### PR TITLE
Generic Radial chart for all start and end angles with open/closed curve options

### PR DESCRIPTION
**Pull request by shaswat-indian**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently the radial chart only renders a complete circle.

Issue Number: N/A

## What is the new behavior?
It now supports rendering a radial chart of any dimension(sector) by setting the newly added `startAngle` and `endAngle` properties. The chart is rendered from  the `startAngle` to the `endAngle` in the **clockwise direction**.

Additionally, a `isClosedCurve` property has been added if closed interpolation of line/area curve is not required (which may be required in cases other than a full circle).

https://github.com/reaviz/reaviz/assets/33908100/aa4cd079-9cfe-4331-bfb6-681ad130c965

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/common/Tooltip/TooltipArea.tsx [hunk 1]
```diff
@@ -134,23 +147,37 @@ export const TooltipArea = forwardRef<any, Partial<TooltipAreaProps>>(({
   const [prevX, setPrevX] = useState<number>();
   const [prevY, setPrevY] = useState<number>();
   const ref = useRef<SVGRectElement | SVGPathElement | any>();
+  const fullCircleref = useRef<SVGRectElement | SVGPathElement | any>();
```
Comment (amcdnl) [inline src/common/Tooltip/TooltipArea.tsx:150]:
```suggestion
  const fullCircleref = useRef<SVGRectElement | SVGPathElement | any>(null);
```

File: src/RadialAreaChart/RadialAreaChart.tsx [hunk 2]
```diff
@@ -40,6 +40,22 @@ export interface RadialAreaChartProps extends ChartProps {
...
+
+  /**
+   * End angle for the last value.
+   */
+  endAngle?: number;
+
+  /**
+   * Whether the curve should be closed. Set to true by deafult
+   */
+  isClosedCurve: boolean;
```
Comment (amcdnl) [inline src/RadialAreaChart/RadialAreaChart.tsx:57]:
```suggestion
  isClosedCurve?: boolean;
```

**Review by amcdnl — APPROVED** [review]

File: src/common/Tooltip/TooltipArea.tsx [hunk 3]
```diff
@@ -233,7 +260,8 @@ export const TooltipArea = forwardRef<any, Partial<TooltipAreaProps>>(({
     }
 
     // Get the path container element
-    let target = ref.current;
+    // Note that we are using the dummy 'full' circle for alignment
+    let target = fullCircleref.current;
```
Comment (gsteffen1203) [inline src/common/Tooltip/TooltipArea.tsx:264]: Hey team, looks like this broke the tooltip rendering in charts other than the new Radial chart. Setting a fallback value to `ref.current` should fix it 👍 

`let target = fullCircleref.current || ref.current;`

Comment (amcdnl) [reply src/common/Tooltip/TooltipArea.tsx:264]: Oh snap - thanks for the feedback! Will address shortly.

Comment (amcdnl) [reply src/common/Tooltip/TooltipArea.tsx:264]: Published fix!

**Pull request merged**